test/DurableTask.Sdk.Tests/Grpc/DurableTaskGrpcClientIntegrationTests.cs

```diff
@@ -0,0 +1,151 @@
+    [InlineData(false)]
+    [InlineData(true)]
+    public async Task GetInstanceMetadata_EndToEnd(bool shouldThrow)
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, string instanceId, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().Be(instanceId);
+                metadata.SerializedInput.Should().BeNull();
+                metadata.SerializedOutput.Should().BeNull();
+                metadata.RuntimeStatus.Should().Be(status);
+                metadata.FailureDetails.Should().BeNull();
+            }
+        }
+
+        await using DurableTaskGrpcWorker server = await this.StartAsync();
+        DurableTaskClient client = this.CreateDurableTaskClient();
+
+        string instanceId = await client.ScheduleNewOrchestrationInstanceAsync(OrchestrationName, input: shouldThrow);
+
+        await client.WaitForInstanceStartAsync(instanceId, default);
+        OrchestrationMetadata? metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, OrchestrationRuntimeStatus.Running);
+
+        await client.RaiseEventAsync(instanceId, "event", default);
+        await client.WaitForInstanceCompletionAsync(instanceId, default);
+        metadata = await client.GetInstanceMetadataAsync(instanceId, false);
+        AssertMetadata(metadata!, instanceId, shouldThrow ? OrchestrationRuntimeStatus.Failed : OrchestrationRuntimeStatus.Completed);
+    }
+
+    [Fact]
+    public async Task GetInstances_EndToEnd()
+    {
+        static void AssertMetadata(OrchestrationMetadata metadata, OrchestrationRuntimeStatus status)
+        {
+            metadata.Should().NotBeNull();
+            using (new AssertionScope())
+            {
+                metadata.Name.Should().Be(OrchestrationName);
+                metadata.InstanceId.Should().StartWith("GetInstances_EndToEnd-");
+                metadata.RuntimeStatus.Should().Be(status);
+
+                // InMemoryOrchestrationService always returns these in a query.
+                // metadata.FailureDetails.Should().BeNull();
+                // metadata.SerializedInput.Should().BeNull();
+                // metadata.SerializedOutput.Should().BeNull();
+            }
+        }
+
+        OrchestrationQuery query = new() { InstanceIdPrefix = "GetInstances_EndToEnd" };
+
+        static async Task ForEachAsync(Func<string, Task> func)
+        {
+            await func("GetInstances_EndToEnd-1");
+            await func("GetInstances_EndToEnd-2");
+        }
+
+        await using DurableTaskGrpcWorker server = await this.StartAsync();
+        DurableTaskClient client = this.CreateDurableTaskClient();
+
+        string notIncluded = await client.ScheduleNewOrchestrationInstanceAsync(OrchestrationName, input: false);
```

Comment:
Unused local variable? If it's just to indicate that this instance is distinct from the others, maybe a comment would be better? It's hard to know without context if an unused local variable is a bug or not.

Reply:
Updated the test to use this variable.